clusterloader2/pkg/prometheus/experimental.go

```diff
@@ -75,21 +89,54 @@ func (pc *PrometheusController) trySnapshotPrometheusDisk() (bool, error) {
 		klog.Info(string(s))
 		return true, nil
 	}
-	snapshotName := pdName
+	pc.diskMetadata.name = pdName
+	pc.diskMetadata.zone = zone
+	return true, nil
+}
+
+func (pc *PrometheusController) snapshotPrometheusDiskIfEnabled() error {
+	if enabled, err := pc.isEnabled(); !enabled {
+		return err
+	}
+	// Update cache of Prometheus disk metadata
+	err := wait.Poll(
+		10*time.Second,
+		2*time.Minute,
+		pc.tryRetrievePrometheusDiskMetadata)
+	if pc.diskMetadata.name == "" || pc.diskMetadata.zone == "" {
+		klog.Errorf("Missing zone or PD name, aborting snapshot")
+		klog.Infof("PD name=%s, zone=%s", pc.diskMetadata.name, pc.diskMetadata.zone)
+		return err
+	}
+	// Select snapshot name
+	snapshotName := pc.diskMetadata.name
 	if *prometheusDiskSnapshotName != "" {
 		if err := VerifySnapshotName(*prometheusDiskSnapshotName); err == nil {
 			snapshotName = *prometheusDiskSnapshotName
 		} else {
-			klog.Warningf("Incorrect disk name %v: %v. Using default name: %v", *prometheusDiskSnapshotName, err, pdName)
+			klog.Warningf("Incorrect disk name %v: %v. Using default name: %v", *prometheusDiskSnapshotName, err, snapshotName)
 		}
 	}
-	klog.Infof("Snapshotting PD '%s' into snapshot '%s' in zone '%s'", pdName, snapshotName, zone)
+	// Snapshot Prometheus disk
+	return wait.Poll(
+		10*time.Second,
+		1*time.Minute,
```

Comment:
The gcloud compute disks snapshot can easily take O(minutes) for big disks (e.g. prometheus disks in 5K node clusters), so the timeout should be bigger than 1 min.

Thanks for improving this code!

Reply:
ok, changed to 10min